app/models/ability.rb

```diff
@@ -29,4 +31,12 @@ def initialize(user)
       can :read, ActiveAdmin
     end
   end
+
+  private
+
+  MINIMUM_DAYS_TO_CANCEL = 7
+
+  def within_cancel_range?(vacation)
+    vacation[:status] == :approved && ((vacation.start_date - Date.today).to_i >= MINIMUM_DAYS_TO_CANCEL)
```

Comment:
Since the vacation model is using the `enumerize` gem, you can use the helper methods that the enum add. https://github.com/brainspec/enumerize#basic

And also, you can use some helper methods from date to write a "better" way to check the dates: https://api.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-days_ago

```suggestion
    vacation.approved? && vacation.start_date.days_ago(MINIMUM_DAYS_TO_CANCEL) >= Date.today
```